pymeasure/instruments/hcp/tc038d.py

```diff
@@ -0,0 +1,136 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2022 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+
+from pymeasure.instruments import Instrument
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class TC038D(Instrument):
+    """
+    Communication with the HCP TC038D oven.
+
+    This is the newer version with DC heating.
+
+    The oven expects raw bytes written, no ascii code, and sends raw bytes.
+    For the variables are two or four-byte modes available. We use the
+    four-byte mode addresses, so do we. In that case element count has to be
+    double the variables read.
+    """
+
+    functions = {'read': 0x03, 'writeMultiple': 0x10,
+                 'writeSingle': 0x06, 'echo': 0x08}
+
+    def __init__(self, resourceName, address=1, timeout=1000):
+        """Initialize the device."""
+        super().__init__(resourceName, "TC038D", timeout=timeout)
+        self.address = address
+
+    def CRC16(self, data):
```

Comment:
This could be generally useful; it would be good to have some tests that document expected input/output of this function. Doesn't CRC have different "variants" that could slightly change outcomes? If so, which one is that?
btw, do you know about Python's `binascii.crc_hqx()`?

Reply:
Yeah, finding the right variant is not easy. Therefore I had chosen to implement a method according to the manuals instruction (which describes a step by step procedure), that way I know, that it will work.

Reply:
It would be good to compare the output of your function with `binascii.crc_hqx()`. For that (and generally), it would be useful if your `CRC16` were a stand-alone function, it does not use the class' state (i.e. `self`), anyway.
Either extract it to method level (preferred), or add a ` @staticmethod` decorator so you can run it without creating a `TC038` instance first. 

If behaviour is identical you can use that one from the Python standard library, instead.

Reply:
OK, don't worry about it then.